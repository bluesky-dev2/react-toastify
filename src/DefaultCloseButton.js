/* eslint react/require-default-props: 0 */
import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'glamor';

const styles = isDefault =>
  css({
    color: isDefault ? '#000' : '#fff',
    fontWeight: 'bold',
    fontSize: '14px',
    background: 'transparent',
    outline: 'none',
    border: 'none',
    padding: 0,
    cursor: 'pointer',
    opacity: isDefault ? '0.3' : '0.7',
    transition: '.3s ease',
    alignSelf: 'flex-start',
    ':hover, :focus': {
      opacity: 1
    }
  });

function DefaultCloseButton({ closeToast, type, ariaLabel }) {
  return (
    <button
      {...styles(type === 'default')}
      type="button"
      onClick={closeToast}
      aria-label={ariaLabel}
    >
      âœ–
    </button>
  );
}

DefaultCloseButton.propTypes = {
  closeToast: PropTypes.func,
  arialLabel: PropTypes.string
};

DefaultCloseButton.defaultProps = {
  ariaLabel: 'close'
};

export default DefaultCloseButton;
