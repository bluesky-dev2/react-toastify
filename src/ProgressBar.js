import React from 'react';
import PropTypes from 'prop-types';
import { css } from "glamor";

import { TYPE } from './constant';
import style from "./style";

const trackProgress = css.keyframes('track-progress', {
  "0%": { width: "100%" },
  "100%": { width: 0 }
});

const progress = type => css({
  position: "absolute",
  bottom: 0,
  left: 0,
  width: 0,
  height: "5px",
  zIndex: 999,
  opacity: 0.7,
  animation: `${trackProgress} linear 1`,
  backgroundColor: "rgba(255,255,255,.7)",
  ...type === "default" ? { background: style.colorProgressDefault } : {}
});

function ProgressBar({ delay, isRunning, closeToast, type, hide, className }) {
  const style = {
    animationDuration: `${delay}ms`,
    animationPlayState: isRunning ? 'running' : 'paused'
  };
  style.WebkitAnimationPlayState = style.animationPlayState;

  if (hide) {
    style.opacity = 0;
  }

  return (
    <div
      {...progress(type)}
      className={className}
      style={style}
      onAnimationEnd={closeToast}
    />
  );
}

ProgressBar.propTypes = {
  /**
   * The animation delay which determine when to close the toast
   */
  delay: PropTypes.number.isRequired,

  /**
   * Whether or not the animation is running or paused
   */
  isRunning: PropTypes.bool.isRequired,

  /**
   * Func to close the current toast
   */
  closeToast: PropTypes.func.isRequired,

  /**
   * Optional type : info, success ...
   */
  type: PropTypes.string,

  /**
   * Hide or not the progress bar
   */
  hide: PropTypes.bool,

  /**
   * Optionnal className
   */
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
};

ProgressBar.defaultProps = {
  type: TYPE.DEFAULT,
  hide: false,
  className: ''
};

export default ProgressBar;
