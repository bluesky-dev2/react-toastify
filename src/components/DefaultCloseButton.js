/* eslint react/require-default-props: 0 */
import React from 'react';
import PropTypes from 'prop-types';

function DefaultCloseButton({ closeToast, type, ariaLabel }) {
  const className = `react-toastify__close-button react-toastify__close-button--${type}`;

  return (
    <button
      className={className}
      type="button"
      onClick={closeToast}
      aria-label={ariaLabel}
    >
      âœ–
    </button>
  );
}

DefaultCloseButton.propTypes = {
  closeToast: PropTypes.func,
  arialLabel: PropTypes.string
};

DefaultCloseButton.defaultProps = {
  ariaLabel: 'close'
};

export default DefaultCloseButton;
