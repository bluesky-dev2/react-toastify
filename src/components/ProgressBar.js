import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import { TYPE } from './../utils/constant';

function ProgressBar({
  delay,
  isRunning,
  closeToast,
  type,
  hide,
  className,
  rtl
}) {
  const style = {
    animationDuration: `${delay}ms`,
    animationPlayState: isRunning ? 'running' : 'paused',
    opacity: hide ? 0 : 1
  };
  style.WebkitAnimationPlayState = style.animationPlayState;

  const classNames = cx(
    'react-toastify__progress-bar',
    `react-toastify__progress-bar--${type}`,
    className,
    {
      'react-toastify__progress-bar--rtl': rtl
    }
  );

  return (
    <div className={classNames} style={style} onAnimationEnd={closeToast} />
  );
}

ProgressBar.propTypes = {
  /**
   * The animation delay which determine when to close the toast
   */
  delay: PropTypes.number.isRequired,

  /**
   * Whether or not the animation is running or paused
   */
  isRunning: PropTypes.bool.isRequired,

  /**
   * Func to close the current toast
   */
  closeToast: PropTypes.func.isRequired,

  /**
   * Support rtl content
   */
  rtl: PropTypes.bool.isRequired,

  /**
   * Optional type : info, success ...
   */
  type: PropTypes.string,

  /**
   * Hide or not the progress bar
   */
  hide: PropTypes.bool,

  /**
   * Optionnal className
   */
  className: PropTypes.string
};

ProgressBar.defaultProps = {
  type: TYPE.DEFAULT,
  hide: false,
};

export default ProgressBar;
