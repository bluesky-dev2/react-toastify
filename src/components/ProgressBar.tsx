import React, { CSSProperties } from 'react';
import cx from 'classnames';

import { TYPE, RT_NAMESPACE, TypeOptions } from './../utils';

export interface ProgressBarProps {
  /**
   * The animation delay which determine when to close the toast
   */
  delay: number;

  /**
   * Whether or not the animation is running or paused
   */
  isRunning: boolean;

  /**
   * Func to close the current toast
   */
  closeToast: () => void;

  /**
   * Optional type : info, success ...
   */
  type: TypeOptions;

  /**
   * Hide or not the progress bar
   */
  hide?: boolean;

  /**
   * Optionnal className
   */
  className?: string | null;

  /**
   * Optionnal inline style
   */
  style?: CSSProperties;

  /**
   * Tell wether or not controlled progress bar is used
   */
  controlledProgress?: boolean;

  /**
   * Controlled progress value
   */
  progress?: number | string;

  /**
   * Support rtl content
   */
  rtl?: boolean;
}

export function ProgressBar({
  delay,
  isRunning,
  closeToast,
  type,
  hide,
  className,
  style: userStyle,
  controlledProgress,
  progress,
  rtl
}: ProgressBarProps) {
  const style: CSSProperties = {
    ...userStyle,
    animationDuration: `${delay}ms`,
    animationPlayState: isRunning ? 'running' : 'paused',
    opacity: hide ? 0 : 1
  };

  if (controlledProgress) style.transform = `scaleX(${progress})`;

  const classNames = cx(
    `${RT_NAMESPACE}__progress-bar`,
    controlledProgress
      ? `${RT_NAMESPACE}__progress-bar--controlled`
      : `${RT_NAMESPACE}__progress-bar--animated`,
    `${RT_NAMESPACE}__progress-bar--${type}`,
    {
      [`${RT_NAMESPACE}__progress-bar--rtl`]: rtl
    },
    className
  );

  // 🧐 controlledProgress is derived from progress
  // so if controlledProgress is set
  // it means that this is also the case for progress
  const animationEvent = {
    [controlledProgress && progress! >= 1
      ? 'onTransitionEnd'
      : 'onAnimationEnd']:
      controlledProgress && progress! < 1 ? null : closeToast
  };

  return <div className={classNames} style={style} {...animationEvent} />;
}

ProgressBar.defaultProps = {
  type: TYPE.DEFAULT,
  hide: false
};
