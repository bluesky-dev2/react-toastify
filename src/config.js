export default {
  POSITION: {
    TOP_LEFT: 'top-left',
    TOP_RIGHT: 'top-right',
    TOP_CENTER: 'top-center',
    BOTTOM_LEFT: 'bottom-left',
    BOTTOM_RIGHT: 'bottom-right',
    BOTTOM_CENTER: 'bottom-center'
  },
  TYPE: {
    INFO: 'info',
    SUCCESS: 'success',
    WARNING: 'warning',
    ERROR: 'error',
    DEFAULT: 'default'
  },
  ACTION: {
    SHOW: 'SHOW_TOAST',
    CLEAR: 'CLEAR_TOAST',
    MOUNTED: 'CONTAINER_MOUNTED'
  }
};
