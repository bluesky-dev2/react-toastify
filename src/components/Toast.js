import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import ProgressBar from './ProgressBar';
import { POSITION, TYPE } from './../utils/constant';
import {
  falseOrElement,
  falseOrDelay,
  objectValues
} from './../utils/propValidator';

class Toast extends Component {
  static propTypes = {
    closeButton: falseOrElement.isRequired,
    autoClose: falseOrDelay.isRequired,
    children: PropTypes.node.isRequired,
    closeToast: PropTypes.func.isRequired,
    position: PropTypes.oneOf(objectValues(POSITION)).isRequired,
    pauseOnHover: PropTypes.bool.isRequired,
    closeOnClick: PropTypes.bool.isRequired,
    transition: PropTypes.func.isRequired,
    isDocumentHidden: PropTypes.bool.isRequired,
    rtl: PropTypes.bool.isRequired,
    in: PropTypes.bool,
    onExited: PropTypes.func,
    hideProgressBar: PropTypes.bool,
    onOpen: PropTypes.func,
    onClose: PropTypes.func,
    type: PropTypes.oneOf(objectValues(TYPE)),
    className: PropTypes.string,
    bodyClassName: PropTypes.string,
    progressClassName: PropTypes.string,
    updateId: PropTypes.number,
    ariaLabel: PropTypes.string
  };

  static defaultProps = {
    type: TYPE.DEFAULT,
    in: true,
    hideProgressBar: false,
    onOpen: null,
    onClose: null,
    className: null,
    bodyClassName: null,
    progressClassName: null,
    updateId: null,
    role: 'alert'
  };

  state = {
    isRunning: true
  };

  componentDidMount() {
    this.props.onOpen !== null && this.props.onOpen(this.getChildrenProps());
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.isDocumentHidden !== nextProps.isDocumentHidden) {
      this.setState({
        isRunning: !nextProps.isDocumentHidden
      });
    }
  }

  componentWillUnmount() {
    this.props.onClose !== null && this.props.onClose(this.getChildrenProps());
  }

  getChildrenProps() {
    return this.props.children.props;
  }

  getToastProps() {
    const toastProps = {};

    if (this.props.autoClose !== false && this.props.pauseOnHover === true) {
      toastProps.onMouseEnter = this.pauseToast;
      toastProps.onMouseLeave = this.playToast;
    }
    typeof this.props.className === 'string' &&
      (toastProps.className = this.props.className);
    this.props.closeOnClick && (toastProps.onClick = this.props.closeToast);

    return toastProps;
  }

  pauseToast = () => {
    this.setState({ isRunning: false });
  };

  playToast = () => {
    this.setState({ isRunning: true });
  };

  render() {
    const {
      closeButton,
      children,
      autoClose,
      type,
      hideProgressBar,
      closeToast,
      transition: Transition,
      position,
      onExited,
      className,
      bodyClassName,
      progressClassName,
      updateId,
      role,
      rtl
    } = this.props;

    const toastClassname = cx(
      'Toastify__toast',
      `Toastify__toast--${type}`,
      className,
      {
        'Toastify__toast--rtl': rtl
      }
    );

    const bodyClassname = cx('Toastify__toast-body', bodyClassName);

    return (
      <Transition
        in={this.props.in}
        appear
        unmountOnExit
        onExited={onExited}
        position={position}
      >
        <div
          className={toastClassname}
          {...this.getToastProps()}
        >
          <div
            {...this.props.in && { role: role }}
            className={bodyClassname}
          >
            {children}
          </div>
          {closeButton !== false && closeButton}
          {autoClose !== false && (
            <ProgressBar
              key={`pb-${updateId}`}
              rtl={rtl}
              delay={autoClose}
              isRunning={this.state.isRunning}
              closeToast={closeToast}
              hide={hideProgressBar}
              type={type}
              className={progressClassName}
            />
          )}
        </div>
      </Transition>
    );
  }
}

export default Toast;
