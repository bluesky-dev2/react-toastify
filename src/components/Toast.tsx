import * as React from 'react';
import cx from 'clsx';

import { ProgressBar } from './ProgressBar';
import { ToastProps } from '../types';
import { Default, isFn } from '../utils';
import { useToast } from '../hooks';

export const Toast: React.FC<ToastProps> = props => {
  const {
    isRunning,
    preventExitTransition,
    toastRef,
    eventHandlers
  } = useToast(props);
  const {
    closeButton,
    children,
    autoClose,
    onClick,
    type,
    hideProgressBar,
    closeToast,
    transition: Transition,
    position,
    className,
    style,
    bodyClassName,
    bodyStyle,
    progressClassName,
    progressStyle,
    updateId,
    role,
    progress,
    rtl,
    toastId,
    deleteToast,
    isIn
  } = props;
  const defaultClassName = cx(
    `${Default.CSS_NAMESPACE}__toast`,
    `${Default.CSS_NAMESPACE}__toast--${type}`,
    {
      [`${Default.CSS_NAMESPACE}__toast--rtl`]: rtl
    }
  );
  const cssClasses = isFn(className)
    ? className({
        rtl,
        position,
        type,
        defaultClassName
      })
    : cx(defaultClassName, className);
  const isProgressControlled = !!progress;

  function renderCloseButton(closeButton: any) {
    if (!closeButton) return;

    const props = { closeToast, type };

    if (isFn(closeButton)) return closeButton(props);

    if (React.isValidElement(closeButton))
      return React.cloneElement(closeButton, props);
  }

  return (
    <Transition
      isIn={isIn}
      done={deleteToast}
      position={position}
      preventExitTransition={preventExitTransition}
      nodeRef={toastRef}
    >
      <div
        id={toastId as string}
        onClick={onClick}
        className={cssClasses}
        {...eventHandlers}
        style={style}
        ref={toastRef}
      >
        <div
          {...(isIn && { role: role })}
          className={
            isFn(bodyClassName)
              ? bodyClassName({ type })
              : cx(`${Default.CSS_NAMESPACE}__toast-body`, bodyClassName)
          }
          style={bodyStyle}
        >
          {children}
        </div>
        {renderCloseButton(closeButton)}
        {(autoClose || isProgressControlled) && (
          <ProgressBar
            {...(updateId && !isProgressControlled
              ? { key: `pb-${updateId}` }
              : {})}
            rtl={rtl}
            delay={autoClose as number}
            isRunning={isRunning}
            isIn={isIn}
            closeToast={closeToast}
            hide={hideProgressBar}
            type={type}
            style={progressStyle}
            className={progressClassName}
            controlledProgress={isProgressControlled}
            progress={progress}
          />
        )}
      </div>
    </Transition>
  );
};
