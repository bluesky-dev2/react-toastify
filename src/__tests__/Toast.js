/* eslint-env jest */
import React from 'react';
import { shallow, mount } from 'enzyme';

import Toast from './../Toast';
import DefaultCloseButton from './../DefaultCloseButton';
import ProgressBar from './../ProgressBar';
import config from './../config';

const REQUIRED_PROPS = {
  closeButton: <DefaultCloseButton />,
  autoClose: 5000,
  closeToast: () => {},
  position: config.POSITION.TOP_RIGHT,
  pauseOnHover: true,
  closeOnClick: true
};

describe('Toast', () => {
  it('Should merge container and body className', () => {
    const component = shallow(
      <Toast
        {...REQUIRED_PROPS}
        autoClose={false}
        className="container-class"
        bodyClassName="body-class"
      >
        FooBar
      </Toast>
    );

    expect(component.find('.container-class')).toHaveLength(1);
    expect(component.find('.body-class')).toHaveLength(1);
  });

  it('Should not render ProgressBar if autoClose prop is set to false', () => {
    const component = shallow(
      <Toast
        {...REQUIRED_PROPS}
        autoClose={false}
      >
        FooBar
      </Toast>
    );

    expect(component.children().find(ProgressBar).length).toBe(0);
  });

  it('Should not render closeButton if closeButton prop is set to false',
    () => {
      const component = shallow(
        <Toast
          {...REQUIRED_PROPS}
          closeButton={false}
        >
          FooBar
        </Toast>
      );

      expect(component.children().find(DefaultCloseButton).length).toBe(0);
    });

  it('Can call onOpen callback when component mount', () => {
    const onOpen = jest.fn();
    mount(
      <Toast
        {...REQUIRED_PROPS}
        onOpen={onOpen}
      >
        FooBar
      </Toast>
    );

    expect(onOpen).toHaveBeenCalled();
  });

  it('Can call onClose callback when component will unmount', () => {
    const onClose = jest.fn();
    const component = mount(
      <Toast
        {...REQUIRED_PROPS}
        onClose={onClose}
      >
        FooBar
      </Toast>
    );

    component.unmount();
    expect(onClose).toHaveBeenCalled();
  });

  it('Can pause toast delay on mouse enter', () => {
    const component = shallow(
      <Toast
        {...REQUIRED_PROPS}
      >
        FooBar
      </Toast>
    );

    expect(component.state('isRunning')).toBeTruthy();
    component.simulate('mouseEnter');
    expect(component.state('isRunning')).toBeFalsy();
  });


  it('Can keep runing on mouse enter', () => {
    const component = shallow(
      <Toast
        {...REQUIRED_PROPS}
        pauseOnHover={false}
      >
        FooBar
      </Toast>
    );

    expect(component.state('isRunning')).toBeTruthy();
    component.simulate('mouseEnter');
    expect(component.state('isRunning')).toBeTruthy();
  });

  it('Should play toast delay on mouse leave', () => {
    const component = shallow(
      <Toast
        {...REQUIRED_PROPS}
      >
        FooBar
      </Toast>
    );

    expect(component.state('isRunning')).toBeTruthy();
    component.simulate('mouseEnter');
    expect(component.state('isRunning')).toBeFalsy();
    component.simulate('mouseLeave');
    expect(component.state('isRunning')).toBeTruthy();
  });

  it('Should not call setState if autoClose prop is false', () => {
    const component = shallow(
      <Toast
        {...REQUIRED_PROPS}
        autoClose={false}
      >
        FooBar
      </Toast>
    );

    expect(component.state('isRunning')).toBeTruthy();
    component.simulate('mouseEnter');
    expect(component.state('isRunning')).toBeTruthy();
  });
});
