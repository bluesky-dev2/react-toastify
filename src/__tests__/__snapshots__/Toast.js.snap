// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Should support Rtl display 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Toast
    autoClose={false}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    closeToast={[Function]}
    hideProgressBar={false}
    in={true}
    isDocumentHidden={false}
    onClose={null}
    onOpen={null}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    role="alert"
    rtl={true}
    transition={[Function]}
    type="default"
    updateId={null}
  >
    FooBar
  </Toast>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {
      "appear": true,
      "children": <div
        className="Toastify__toast Toastify__toast--default Toastify__toast--rtl"
        onClick={[Function]}
      >
        <div
          className="Toastify__toast-body"
          role="alert"
        >
          FooBar
        </div>
        <CloseButton
          ariaLabel="close"
        />
      </div>,
      "in": true,
      "onExited": undefined,
      "position": "top-right",
      "unmountOnExit": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Toastify__toast-body"
            role="alert"
          >
            FooBar
          </div>,
          <CloseButton
            ariaLabel="close"
          />,
          false,
        ],
        "className": "Toastify__toast Toastify__toast--default Toastify__toast--rtl",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "FooBar",
            "className": "Toastify__toast-body",
            "role": "alert",
          },
          "ref": null,
          "rendered": "FooBar",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "ariaLabel": "close",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "appear": true,
        "children": <div
          className="Toastify__toast Toastify__toast--default Toastify__toast--rtl"
          onClick={[Function]}
        >
          <div
            className="Toastify__toast-body"
            role="alert"
          >
            FooBar
          </div>
          <CloseButton
            ariaLabel="close"
          />
        </div>,
        "in": true,
        "onExited": undefined,
        "position": "top-right",
        "unmountOnExit": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Toastify__toast-body"
              role="alert"
            >
              FooBar
            </div>,
            <CloseButton
              ariaLabel="close"
            />,
            false,
          ],
          "className": "Toastify__toast Toastify__toast--default Toastify__toast--rtl",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "FooBar",
              "className": "Toastify__toast-body",
              "role": "alert",
            },
            "ref": null,
            "rendered": "FooBar",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "ariaLabel": "close",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
